### PR TITLE
[CARBONDATA-3885] [CARBONDATA-3884] Fix for load failures with isSIenabled = false and fix for concurrent load failure

### PR DESCRIPTION
**Pull request by vikramahuja1001**
### Why is this PR needed?
 ### Issue 1: 
In some concurrent scenarios it was found that certain segment folders were getting deleted for the SI table. When the segment was being inserted again back to the SI table during a new load to the main table by doing isSIenabled = false, the load to the SI table failed.

Root Cause: It was found that when the segment folder gets deleted and it's entry from table status file is manually deleted then, during new load to SI when table status is changed to InsertInProgress during the loading previously deleted segments the stale .segmentfile still exists which was pointing to the mergeindexfile which does exist anymore(since the segment folder is deleted). Since that mergeindexfile does not exist anymore the load to SI table failed

### Issue 2: 
In some concurrent load scenarios in main table with SI table with isSITableEnabled = false , load to SI table fails.
 
Root Cause: When concurrent loading is done with isSITableEnabled = false, maintable details are taken earlier in the code and SItable details are taken later. It was found that there were some scenarios where another concurrent load to main table finishes and can change the number of segments in the main table due to which there is mismatch in number of segments in the SI table and in the main table. Due to this mismatch when the validmaintableload method is called, there can be scenarios where the load fails.

 ### What changes were proposed in this PR?
Solution for Issue 1: Whenever isSIenabled is set to false and it tries to load previously deleted SI segments to the SI table we go and delete the stale .segment file from the metadata folder if it exists. Due to this, while loading a new .segment file will be created thus having a new index file path.
    
Solution for Issue 2: Whenever isSIenabled is set to false and it tries to load previously deleted SI segments, just before adding segments in SIFailedSegmentList checking again if the loadname is present in the main table. Only adding in SIFailedSegmenList if the loadname is present in the main table and it's in SUCCESS state.

 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3201/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1475/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3307/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1569/

**Comment by akashrn5** [comment]
@vikramahuja1001 the  issue reproduce steps, please mention in jira and for PR description, give proper issue, root cause and solution proposed.

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/CarbonInternalLoaderUtil.java [hunk 1]
```diff
@@ -37,7 +38,6 @@
 import org.apache.carbondata.core.util.path.CarbonTablePath;
 import org.apache.carbondata.processing.loading.model.CarbonLoadModel;
 import org.apache.carbondata.processing.util.CarbonLoaderUtil;
-
```
[the remark this answers is not in the document]
Comment (vikramahuja1001) [reply integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/CarbonInternalLoaderUtil.java]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3326/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1586/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3351/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1610/

**Comment by VenuReddy2103** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3352/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1611/

**Comment by akashrn5** [comment]
LGTM

**Pull request closed without merging**